vspace/src/main/java/edu/asu/diging/vspace/web/staff/AddImageBlockController.java

```diff
@@ -0,0 +1,52 @@
+package edu.asu.diging.vspace.web.staff;
+
+import java.io.IOException;
+import java.security.Principal;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import edu.asu.diging.vspace.core.services.IContentBlockManager;
+
+public class AddImageBlockController {
+
+    @Autowired
+    private IContentBlockManager contentBlockManager;
+
+    @RequestMapping(value = "/staff/module/slide/{id}/imagecontent", method = RequestMethod.POST)
+    public ResponseEntity<String> addImageBlock(@PathVariable("id") String slideId,
+            @RequestParam("file") MultipartFile file, Principal principal, RedirectAttributes attributes)
+            throws IOException {
+
+        byte[] bgImage = null;
+        String filename = null;
+        if (file.isEmpty() || file.equals(null)) {
```

Comment:
why do you not check for null here? what if file is indeed null? you would get a null pointer here. if file is null, line 43 will never be reached.